### DISCUSSION TITLE
How to compute a field ?

### DISCUSSION THREAD

**Discussion by lucas-garrido** [linq2db/linq2db#3189] [Q&A] [answered]
i would like to use functions like TO_DATE or TRIM on a field, right now i'm doing this :

![image](https://user-images.githubusercontent.com/48165310/131152217-29cd917b-df06-432e-b1c1-fc8374024653.png)
but i would like to do it at SQL level. 
I've tried to use expressionMethod but i failed 😁... could you provide an example please ? 🙏🙏🙏

**Comment by MaceWindu** [comment, marked as answer]
```cs
// computed property
[ExpressionMethod(nameof(CreatedAtExpr), IsColumn = true)]
public DateTime? CreatedAt { get; set; }

// property implementaion
private static Expression<Func<Customer, DateTime?>> CreatedAtExpr() => e => MyFunctions.ToDate(e.CreatedAtRaw);
```

```cs
// TO_DATE function mapping as linq2db has no built-in mapping
public static class MyFunctions
{
    [Function("TO_DATE", ServerSideOnly = true)]
    public static DateTime? ToDate(string value) => throw new InvalidOperationException(); // or provide C# implementation if intended to call THIS method from non-db code too
}
```

**Reply by lucas-garrido** [reply]
Thanks !! it works perfectly ! What if i want to add parameter like TO_DATE( $value ,'YYYYMMDD') ?

**Reply by MaceWindu** [reply]
you have two options:
- for variable parameter - just add additional parameter to method
- for fixed parameter - use `Sql.Expresssion("TO_DATE({0}, 'YOURFORMAT')", ...)` instead of `Sql.Function`
